web-components/src/components/inputs/validation.ts

```diff
@@ -5,6 +5,17 @@ export const invalidPassword: string =
 export const requirementAgreement: string = 'You must agree to continue';
 export const invalidAmount: string = 'Please enter a valid amount';
 export const insufficientCredits: string = `You don't have enough credits`;
+export const invalidDate: string = `Invalid date`;
+export const invalidURL: string = `Please enter a valid URL`;
+export const invalidVCSRetirement: string = `Please enter a valid VCS retirement serial number`;
+export const invalidVCSID: string = `Please enter a valid VCS Project ID`;
+
+export const numericOnlyRE = /^\d*$/gm;
+
+// RegEx based on https://verra.org/wp-content/uploads/2020/09/VCU-Serial-Number-Help-Format.pdf
+// TODO: it is possible to do additional validation to make sure dates and project ID match serial number. See link above.
```

Comment:
Leaving a note here. We could theoretically validate the `startDate`, `endDate`, and `vcsProjectId` against the `vceRetirementSerialNumber` because they are all components of the serial number. It would be complex, but it would help with data integrity. (I think it would be too much to do for now).

Reply:
that seems quite trivial to verify, although I'm fine creating an issue for that and doing that in a follow-up
ideally the user would only have to input the serial number and then startDate, endDate and vcsProjectId would get populated automatically